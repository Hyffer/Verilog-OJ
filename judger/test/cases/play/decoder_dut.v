module decoder(
    input [2:0] x,
    output reg [7:0] y
    );
    integer i;
    always @ (*) begin
        for (i = 0; i < 8; i = i + 1) begin
            if (x == i[2:0]) begin
                y = 8'd0;
                y[i] = 1;
            end
        end
    end
endmodule