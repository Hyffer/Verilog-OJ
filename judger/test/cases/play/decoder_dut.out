#! /usr/bin/vvp
:ivl_version "10.1 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "system";
:vpi_module "vhdl_sys";
:vpi_module "v2005_math";
:vpi_module "va_math";
S_0x55faf9484ff0 .scope module, "testbench" "testbench" 2 1;
 .timescale 0 0;
v0x55faf94a55c0_0 .var/i "i", 31 0;
v0x55faf94a56a0_0 .var "x", 2 0;
v0x55faf94a5790_0 .net "y", 7 0, v0x55faf94a54a0_0;  1 drivers
S_0x55faf9485170 .scope module, "DUT" "decoder" 2 4, 3 1 0, S_0x55faf9484ff0;
 .timescale 0 0;
    .port_info 0 /INPUT 3 "x"
    .port_info 1 /OUTPUT 8 "y"
v0x55faf9485340_0 .var/i "i", 31 0;
v0x55faf94a53c0_0 .net "x", 2 0, v0x55faf94a56a0_0;  1 drivers
v0x55faf94a54a0_0 .var "y", 7 0;
E_0x55faf9480970 .event edge, v0x55faf9485340_0, v0x55faf94a53c0_0;
    .scope S_0x55faf9485170;
T_0 ;
    %wait E_0x55faf9480970;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x55faf9485340_0, 0, 32;
T_0.0 ;
    %load/vec4 v0x55faf9485340_0;
    %cmpi/s 8, 0, 32;
    %jmp/0xz T_0.1, 5;
    %load/vec4 v0x55faf94a53c0_0;
    %load/vec4 v0x55faf9485340_0;
    %parti/s 3, 0, 2;
    %cmp/e;
    %jmp/0xz  T_0.2, 4;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v0x55faf94a54a0_0, 0, 8;
    %pushi/vec4 1, 0, 1;
    %ix/getv/s 4, v0x55faf9485340_0;
    %store/vec4 v0x55faf94a54a0_0, 4, 1;
T_0.2 ;
    %load/vec4 v0x55faf9485340_0;
    %addi 1, 0, 32;
    %store/vec4 v0x55faf9485340_0, 0, 32;
    %jmp T_0.0;
T_0.1 ;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_0x55faf9484ff0;
T_1 ;
    %vpi_call 2 8 "$dumpfile", "out.vcd" {0 0 0};
    %vpi_call 2 17 "$dumpvars", 32'sb00000000000000000000000000000000, v0x55faf94a56a0_0, v0x55faf94a5790_0 {0 0 0};
    %end;
    .thread T_1;
    .scope S_0x55faf9484ff0;
T_2 ;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x55faf94a55c0_0, 0, 32;
T_2.0 ;
    %load/vec4 v0x55faf94a55c0_0;
    %cmpi/s 8, 0, 32;
    %jmp/0xz T_2.1, 5;
    %delay 1, 0;
    %load/vec4 v0x55faf94a55c0_0;
    %pad/s 3;
    %store/vec4 v0x55faf94a56a0_0, 0, 3;
    %load/vec4 v0x55faf94a55c0_0;
    %addi 1, 0, 32;
    %store/vec4 v0x55faf94a55c0_0, 0, 32;
    %jmp T_2.0;
T_2.1 ;
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "testbench.v";
    "decoder_dut.v";
